database_client/database_client.py

```diff
@@ -573,54 +563,44 @@ def get_user_info(self, email: str) -> UserInfo:
         :raise UserNotFoundError: If no user is found.
         :return: UserInfo namedtuple containing the user's information.
         """
-        query = sql.SQL(
-            "select id, password_digest, api_key, email from users where email = {email}"
-        ).format(email=sql.Literal(email))
-        self.cursor.execute(query)
-        results = self.cursor.fetchone()
+        query = select(User.id, User.password_digest, User.api_key, User.email).where(
+            User.email == email
+        )
+        results = self.session.execute(query).mappings().one_or_none()
         if results is None:
             raise UserNotFoundError(email)
 
         return self.UserInfo(
-            id=results[0],
-            password_digest=results[1],
-            api_key=results[2],
-            email=results[3],
+            id=results.id,
+            password_digest=results.password_digest,
+            api_key=results.api_key,
+            email=results.email,
         )
 
     def get_user_info_by_external_account_id(
-            self,
-            external_account_id: str,
-            external_account_type: ExternalAccountTypeEnum
+        self, external_account_id: str, external_account_type: ExternalAccountTypeEnum
     ) -> UserInfo:
-        query = sql.SQL("""
-            SELECT 
-                u.id,
-                u.email,
-                u.password_digest,
-                u.api_key
-            FROM 
-                users u
-            INNER JOIN 
-                external_accounts ea ON u.id = ea.user_id
-            WHERE 
-                ea.account_identifier = {external_account_identifier}
-                and ea.account_type = {external_account_type}
-        """).format(
-            external_account_identifier=sql.Literal(external_account_id),
-            external_account_type=sql.Literal(external_account_type.value)
+        u = aliased(User)
+        ea = aliased(ExternalAccount)
+
+        query = (
+            select(u.id, u.email, u.password_digest, u.api_key)
+            .join(ea, u.id == ea.user_id)
+            .where(
+                ea.account_identifier == external_account_id,
+                ea.account_type == external_account_type.value,
+            )
         )
-        self.cursor.execute(query)
+        results = self.session.execute(query).mappings().one_or_none()
 
-        results = self.cursor.fetchone()
         if results is None:
             raise UserNotFoundError(external_account_id)
 
         return self.UserInfo(
-            id=results["id"],
-            password_digest=results["password_digest"],
-            api_key=results["api_key"],
-            email=results["email"],
+            id=results.id,
+            password_digest=results.password_digest,
+            api_key=results.api_key,
+            email=results.email,
```

Comment:
Considering this returning of the `UserInfo` is a 1:1 match for the SQLAlchemy object, it does raise the question as to whether it'd be better to simply return the SQLAlchemy object. Such a move would technically couple it more with the rest of the code, however, so it's worth debating.

Reply:
@maxachis I believe it's worth noting that the `mappings()` method is what is packaging the result as a `dict`. Without it, the results default to a regular list. I'm not yet sure how it will behave when gathering results from two places with the same column name, such as the `name` column from both `data_sources` and `agencies`. But you do make a valid point